src/ts/components/errormessageoverlay.ts

```diff
@@ -142,4 +147,30 @@ export class ErrorMessageOverlay extends Container<ErrorMessageOverlayConfig> {
     // Canvas rendering must be explicitly stopped, else it just continues forever and hogs resources
     this.tvNoiseBackground.stop();
   }
-}
\ No newline at end of file
+}
+
+function customizeErrorMessage(
+  errorMessages: ErrorMessageTranslator | ErrorMessageMap,
+  event: ErrorEvent | MobileV3PlayerErrorEvent | MobileV3SourceErrorEvent,
+): string | undefined {
+  let message = undefined;
+  // Process message vocabularies
+  if (errorMessages) {
+    if (typeof errorMessages === 'function') {
+      // Translation function for all errors
+      message = errorMessages(event);
+    } else if (errorMessages[event.code]) {
+      // It's not a translation function, so it must be a map of strings or translation functions
+      let customMessage = errorMessages[event.code];
+
+      if (typeof customMessage === 'string') {
+        message = customMessage;
+      } else {
+        // The message is a translation function, so we call it
+        message = customMessage(event);
+      }
+    }
+  }
+
+  return message;
+}
```

Comment:
Maybe extract the "check if it's a function and do something, else do something else" logic out?

Reply:
I refactored the function to just have one level of complexity, so we don't need to extract anything 😁